src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -450,20 +435,20 @@ public AddBlockResult SuggestBlock(Block block, BlockTreeSuggestOptions options
         public BlockHeader? FindHeader(long number, BlockTreeLookupOptions options)
         {
             Keccak blockHash = GetBlockHashOnMainOrBestDifficultyHash(number);
-            return blockHash is null ? null : FindHeader(blockHash, options);
+            return blockHash is null ? null : FindHeader(blockHash, options, blockNumber: number);
```

Comment:
you might not need a hash lookup here - if there is only one header with this number.

Reply:
Not sure what you mean, it need both the blocknumber and the hash. Single lookup.

Reply:
Ah right. Hash is still needed with the second one. It could be that the header is not canonical.